### PR TITLE
[API-1357] Added count method

### PR DESCRIPTION
**Pull request by vladimir-trifonov**
https://jira.appcelerator.org/browse/API-1357

### CODE REVIEW

File: lib/methods/query.js [hunk 1]
```diff
@@ -53,10 +53,10 @@ exports.query = function (Model, options, callback) {
 		orderQuery = orderQuery.slice(0, -1);
 	}
 
-	pagingQuery += ' ' + (whereQuery ? 'AND' : 'WHERE') + ' ROWNUM >= ' + (+options.skip || 0) + ' AND ROWNUM <= ' + (+options.limit);
+	pagingQuery += ' OFFSET ' + (+options.skip || 0) + ' ROWS FETCH NEXT ' + (+options.limit || 10) + ' ROWS ONLY';
 
-	var query = 'SELECT ' + keys + ' FROM ' + table + whereQuery + pagingQuery + orderQuery;
-	this._query(query, values, callback, function (rows, results) {
+	var query = 'SELECT ' + keys + ' FROM ' + table + whereQuery + orderQuery + pagingQuery;
+	this._query(query, values, callback, function (rows) {
```
Comment (janvennemann) [inline lib/methods/query.js]: Shouldn't we pass useResultSet here? The limit option in the pagingQuery allows us to specify more than the default of 1000 for maxRows which would result in unexpected behaviour.

Comment (vladimir-trifonov) [inline lib/methods/query.js]: Yes, you're right. I've totally missed it.

**Comment by vladimir-trifonov** [comment]
Added 'useResultSet' option in query.js.

**Comment by janvennemann** [comment]
The unit tests do not run causing npm test to fail. The tests in `test/capabilities/count.js` will not be executed because the count method is not considered to belong to any capability. Could you please add a manual replacement test in `test/connector.js` for now?

**Comment by vladimir-trifonov** [comment]
done!

**Comment by janvennemann** [comment]
CR and FT passed, approved!

**Pull request merged**